proposals/calls-to-action-for-better-patch-adoption.md

```diff
@@ -0,0 +1,126 @@
+>
+> To evaluate precisely the need and the urgency of a library update, it is necessary to answer the key question: “is the vulnerability exploitable, given the particular way the library is used within the application?”. Answering this question is extremely difficult: vulnerabilities are typically described in advisories that consist of short, high-level, textual descriptions in natural language, whereas a reliable assessment of the exploitability and the potential impact of a vulnerability demands much lower-level, detailed, technical information.
+
+- On the Threat of npm Vulnerable Dependencies in Node.js Applications (2020)
+
+https://arxiv.org/abs/2009.09019
+
+> Our findings show that although 67.93% of the examined applications depend on at least one vulnerable package, 94.91% of the vulnerable packages in those affected applications are classified as having low threat. Moreover, we find that in the case of vulnerable packages classified as having high threat, it is the application's lack of updating that makes them vulnerable, i.e., it is not the existence of the vulnerability that is the real problem.
+
+- Out of sight, out of mind? How vulnerable dependencies affect open-source projects (2021)
+
+https://link.springer.com/article/10.1007/s10664-021-09959-3
+
+> We found that project activity level, popularity, and developer experience do not translate into better or worse handling of dependency vulnerabilities. We found that most dependency vulnerabilities persist throughout the observation period (mean of 78.4%, 97.7%, and 66.4% for publicly-known vulnerabilities in our Java, Python, and Ruby datasets respectively), and the resolved ones take 3-5 months to fix.
+
+3. **How is this proposal innovative -- what distinguishes it from other related work?** 
+
+This proposal is innovative in that it is a precursor to direct action. Asking the question of how we can improve adoption of security-related patches for open source libraries begs for resolution in the form of work to implement the findings.
+
+This is not academic research. It is groundwork for development.
+
+4. **Who is your doer -- who will execute the proposed work?** <!-- Insert answer here -->
+  * Lucas Gonze
+    - email: [lucas@gonze.com](mailto:lucas@gonze.com)
+    - CV: [gonze.com/cv.pdf](https://gonze.com/tmp/LucasGonze-CV-OSS_virtual_incubator.pdf)
+
+5. **How might this work be sustained long-term after an initial seed grant?** <!-- Insert answer here -->
+  * The work is a complete deliverable in itself. Follow-up work will be to put identified remediations into action. The follow-up work can be performed independently by the doer and others. 
```

Comment:
How are you thinking of publishing this work? A research paper? A white paper / PDF report? Some other method?

Reply:
To publish it I would target the publishers of comparable research. For example, [this relevant paper](https://dl.acm.org/doi/abs/10.1145/3239235.3268920) is in the ACM Digital Library via Proceedings of the 12th ACM/IEEE International Symposium on Empirical Software Engineering and Measurement. Part of that would surely be conference submissions. In addition I would share the research among the security hacker community, for example in OpenSSF calls.